internal/host/refresh_status_preprocessor.go

```diff
@@ -37,12 +38,13 @@ type refreshPreprocessor struct {
 	disabledHostValidations DisabledHostValidations
 }
 
-func newRefreshPreprocessor(log logrus.FieldLogger, hwValidatorCfg *hardware.ValidatorCfg, hwValidator hardware.Validator, operatorsApi operators.API, disabledHostValidations DisabledHostValidations) *refreshPreprocessor {
+func newRefreshPreprocessor(log logrus.FieldLogger, hwValidatorCfg *hardware.ValidatorCfg, hwValidator hardware.Validator, operatorsApi operators.API, disabledHostValidations DisabledHostValidations, metricsAPI metrics.API) *refreshPreprocessor {
```

Comment:
I am not sure that refreshStatusPreprocessor is the correct location to measure latency metrics.  It can be called every 10s when there is no change occurred between subsequent calls, and it may be called every 5 minutes when many connectivity changes occurred in-between.  
Why not measure in the location that connectivity is received and processed here https://github.com/openshift/assisted-service/blob/8758e8197b6bdd44bfa2e652f344ed6c8e3485fe/internal/host/host.go#L536 ?

Reply:
Seems that the correct place to measure latency is when getting the results, no?

Reply:
Right.  That's the function above

Reply:
Makes sense. thanks for the suggestion.

Reply:
At the end I added it here:
https://github.com/openshift/assisted-service/blob/98f8f74399ded5f9ad68c58ada0a3878fce8ab6d/internal/cluster/transition.go#L533-L534
Let me know what you think.

Reply:
I think this https://github.com/openshift/assisted-service/blob/8758e8197b6bdd44bfa2e652f344ed6c8e3485fe/internal/host/host.go#L536 location is better because it is the place that new connectivity report is handled.
In addition, it should be handled at host level.

Reply:
My concern is about potential re-processing of the connectivity report before the cluster is created due to changes in the cluster topology during pre-installation. Capturing the report only once right before installation gives a clearer picture of these values.